packages/suite/src/actions/wallet/signVerifyActions.ts

```diff
@@ -58,50 +84,72 @@ export const verify = (address: string, message: string, signature: string, hex
     dispatch: Dispatch,
     getState: GetState,
 ) => {
-    const selectedDevice = getState().suite.device;
-    if (!selectedDevice) return;
-    const error = validateAddress(address);
+    const { device } = getState().suite;
+    const { account } = getState().wallet.selectedAccount;
+    if (!device || !account) return;
+
+    let fn;
+    const params = {
+        address,
+        message,
+        signature,
+        coin: account.symbol,
+        hex,
+    };
+    const error = isAddressValid(address, params.coin as Account$['symbol'])
+        ? null
+        : 'TR_ADDRESS_IS_NOT_VALID';
+
+    switch (account.networkType) {
+        case 'bitcoin':
+            fn = TrezorConnect.verifyMessage;
+            break;
+        case 'ethereum':
+            fn = TrezorConnect.ethereumVerifyMessage;
+            break;
+        default:
+            fn = () => ({
+                success: false,
+                payload: {
+                    error: `Unsupported network: ${account.networkType}`,
+                    code: undefined,
+                    signature: '',
+                },
+            });
+            break;
+    }
 
     if (error) {
```

Comment:
move this block to line ~102, before you declare fn and params

Reply:
add 2, is this double-validation needed at here? you've already validate address in `inputChange` therefore you shouldn't be able to call `verify` with invalid address

Reply:
double validation is ok in finance, but ok, I rewritten it